### ISSUE TITLE
accuracy level on 50%

### ISSUE THREAD

**Issue by alexionby** [apache/incubator-mxnet#3656]
I tried to learn mxnet on my data, but achieved only 50%.
This is my arch:

```
X = np.reshape(train, (train.shape[0], 1, kernel_size, kernel_size))
Y = label
print(X.shape, Y.shape)

(trainData, testData, trainLabels, testLabels) = train_test_split(
    X / 255.0, Y.astype("int"), test_size=0.1)

data = mx.symbol.Variable('data')

conv1 = mx.symbol.Convolution(data=data, kernel=(6, 6), num_filter=48)
relu1 = mx.symbol.Activation(data=conv1, act_type="relu")
pool1 = mx.symbol.Pooling(data=relu1, pool_type="max",
                          kernel=(2, 2), stride=(2, 2))
# second conv
conv2 = mx.symbol.Convolution(data=pool1, kernel=(5, 5), num_filter=48)
relu2 = mx.symbol.Activation(data=conv2, act_type="relu")
pool2 = mx.symbol.Pooling(data=relu2, pool_type="max",
                          kernel=(2, 2), stride=(2, 2))
# first fullc
conv3 = mx.symbol.Convolution(data=pool2, kernel=(4, 4), num_filter=48)
relu3 = mx.symbol.Activation(data=conv3, act_type="relu")
pool3 = mx.symbol.Pooling(data=relu3, pool_type="max",
                          kernel=(2, 2), stride=(2, 2))

conv4 = mx.symbol.Convolution(data=pool3, kernel=(2, 2), num_filter=48)
relu4 = mx.symbol.Activation(data=conv4, act_type="relu")
pool4 = mx.symbol.Pooling(data=relu4, pool_type="max",
                          kernel=(2, 2), stride=(2, 2))

flatten = mx.symbol.Flatten(data=pool4)
fc1 = mx.symbol.FullyConnected(data=flatten, num_hidden=100)
relu5 = mx.symbol.Activation(data=fc1, act_type="relu")
# second fullc
fc2 = mx.symbol.FullyConnected(data=relu5, num_hidden=2)
# loss
convnet = mx.symbol.SoftmaxOutput(data=fc2, name='softmax')


train = mx.io.NDArrayIter(
    data=trainData,
    label=trainLabels,
    batch_size=100)

val = mx.io.NDArrayIter(
    data=testData,
    label=testLabels,
    batch_size=100)


logging.basicConfig(level=logging.DEBUG)

model = mx.model.FeedForward(
    ctx=mx.gpu(),
    symbol=convnet,
    num_epoch=10,
    optimizer = 'sgd',
    learning_rate=0.01)

model.fit(
    X=train,
    eval_data=val,
    eval_metric='acc',
    batch_end_callback=mx.callback.Speedometer(100))` 
```

Then I tried to do the same on the another framework _Keras_ ( _tensorFlow_ as backend).
And achieved 75% accuracy after 10 epochs on the **same data**.

There is the arch:
`    

```
X = np.reshape(train,(train.shape[0],1,kernel_size,kernel_size))
Y = np.reshape(label,(label.shape[0],1))
print(X.shape, Y.shape)

print("[INFO] compiling model...")
opt = SGD(lr=0.01)

depth = 1 
height = 65
width = 65
classes = 2

    # initialize the model
model = Sequential()

# first set of CONV => RELU => POOL
model.add(Convolution2D(48, 6, 6, border_mode="same",
                        input_shape=(depth, height, width)))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))

# second set of CONV => RELU => POOL
model.add(Convolution2D(48, 5, 5, border_mode="same"))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))

# second set of CONV => RELU => POOL
model.add(Convolution2D(48, 4, 4, border_mode="same"))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))

# second set of CONV => RELU => POOL
model.add(Convolution2D(48, 2, 2, border_mode="same"))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))

# set of FC => RELU layers
model.add(Flatten())
model.add(Dense(100))
model.add(Activation("relu"))

# softmax classifier
model.add(Dense(classes))
model.add(Activation("softmax"))


model.compile(loss="categorical_crossentropy", optimizer=opt, metrics=["accuracy"])

(trainData, testData, trainLabels, testLabels) = train_test_split( 
    X / 255.0, Y.astype("int"), test_size=0.1)

trainLabels = np_utils.to_categorical(trainLabels, 2)
testLabels = np_utils.to_categorical(testLabels, 2)

print("[INFO] training...")
model.fit(trainData, trainLabels, batch_size=128, nb_epoch=10, verbose=1)

# show the accuracy on the testing set
print("[INFO] evaluating...")
(loss, accuracy) = model.evaluate(testData, testLabels,
                                  batch_size=128, verbose=1)
print("[INFO] accuracy: {:.2f}%".format(accuracy * 100))`
```

But i rly need mxnet, so my question is: "What am i doing wrong?"

**Comment by piiswrong** [comment]
you can debug with examples/python-howto/monitor_weight.py
My guess is try tuning learning rate or batchsize a little

**Comment by EGQM** [comment]
what is the kind of your GPU?

**Comment by alexionby** [comment]
NVIDIA GTX 950

**Comment by EGQM** [comment]
@alexionby 
I think it‘s ok.
some GPUs can cause the  problem.

**Comment by mli** [comment]
add `shuffle=True` in NDArrayIter?

**Comment by alexionby** [comment]
Yep, if i don't, it stay on 1.00 .

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
